File: src/backend/seed.js
const { PrismaClient } = require('@prisma/client');
const prisma = new PrismaClient();

async function main() {
    console.log('🌱 Starting seed...');

    // 1. Create a default user (Integrator)
    const user = await prisma.user.create({
        data: {
            email: 'john@quarks.solar',
            name: 'John Doe',
            password: 'hashed_password_mock', // secure in prod
            role: 'INTEGRADOR'
        }
    });

    console.log(`👤 Created user: ${user.name}`);

    // 2. Create Leads (Spread across Kanban columns)
    const leadsData = [
        { name: 'Fazenda Bom Retiro', status: 'NEW', consumption: 1200, location: 'Ribeirão Preto, SP', value: 145000 },
        { name: 'Indústria MetalMax', status: 'CONTACTED', consumption: 4500, location: 'Joinville, SC', value: 450000 },
        { name: 'Mercado Central', status: 'PROPOSAL_SENT', consumption: 850, location: 'Vitória, ES', value: 32500 },
        { name: 'Residência Silva', status: 'NEGOTIATION', consumption: 450, location: 'Campinas, SP', value: 18000 },
        { name: 'AgroTech Sul', status: 'CLOSED_WON', consumption: 2100, location: 'Londrina, PR', value: 210000 },
        { name: 'Padaria do João', status: 'CLOSED_LOST', consumption: 600, location: 'São Paulo, SP', value: 25000 },
        { name: 'Condomínio Solar', status: 'NEW', consumption: 3500, location: 'Salvador, BA', value: 380000 },
        { name: 'Oficina Mecânica', status: 'CONTACTED', consumption: 550, location: 'Belo Horizonte, MG', value: 22000 },
    ];

    for (const lead of leadsData) {
        const createdLead = await prisma.lead.create({
            data: {
                name: lead.name,
                email: `contact@${lead.name.toLowerCase().replace(/\s/g, '')}.com`,
                consumption: lead.consumption,
                location: lead.location,
                status: lead.status, // Using the enum mapped string
                ownerId: user.id
            }
        });

        // 3. Create Proposal for relevant leads
        if (['PROPOSAL_SENT', 'NEGOTIATION', 'CLOSED_WON'].includes(lead.status)) {
            await prisma.proposal.create({
                data: {
                    title: `Proposta Solar - ${lead.name}`,
                    generationKwh: lead.consumption * 1.1, // 110% offset
                    systemSizeKwp: (lead.consumption / 120), // rough calc
                    totalPrice: lead.value,
                    paybackYears: 3.5,
                    savingsMonthly: lead.consumption * 0.95, // tariff mock
                    status: lead.status === 'CLOSED_WON' ? 'ACCEPTED' : 'SENT',
                    leadId: createdLead.id,
                    creatorId: user.id
                }
            });
        }
    }

    console.log('✅ Seed complete!');
}

main()
    .catch((e) => {
        console.error(e);
        process.exit(1);
    })
    .finally(async () => {
        await prisma.$disconnect();
    });

File: src/calc_engine/seed.py
import asyncio
import sys
import os

# Fix import path
sys.path.append(os.getcwd())

from src.prisma import prisma
from prisma.enums import LeadStatus, Role

async def main():
    await prisma.connect()
    
    # Create User if not exists
    user = await prisma.user.find_first(where={"email": "admin@quarks.com"})
    if not user:
        user = await prisma.user.create(
            data={
                "email": "admin@quarks.com",
                "password": "hashed_secret", # Mock
                "name": "Admin User",
                "role": Role.ADMIN
            }
        )
        print("Created Admin User")
    
    # Create Leads
    leads_data = [
        {"name": "Padaria do João", "consumption": 1200.0, "status": LeadStatus.NEW, "distributor": "ENEL_SP"},
        {"name": "Residência Silva", "consumption": 450.0, "status": LeadStatus.CONTACTED, "distributor": "CPFL_PAULISTA"},
        {"name": "Mercado Central", "consumption": 3500.0, "status": LeadStatus.PROPOSAL_SENT, "distributor": "CEMIG"},
        {"name": "Sítio Boa Vista", "consumption": 800.0, "status": LeadStatus.NEGOTIATION, "distributor": "ELEKTRO"},
        {"name": "Indústria Tech", "consumption": 15000.0, "status": LeadStatus.CLOSED_WON, "distributor": "ENEL_SP"},
    ]
    
    for lead in leads_data:
        exists = await prisma.lead.find_first(where={"name": lead["name"]})
        if not exists:
            await prisma.lead.create(
                data={
                    "name": lead["name"],
                    "consumption": lead["consumption"],
                    "status": lead["status"],
                    "distributor": lead["distributor"],
                    "ownerId": user.id
                }
            )
            print(f"Created Lead: {lead['name']}")
    
    await prisma.disconnect()

if __name__ == "__main__":
    asyncio.run(main())
